x-pack/solutions/security/plugins/security_solution/common/constants.ts

```diff
@@ -287,6 +283,12 @@ export const NOTE_URL = '/api/note' as const;
 export const PINNED_EVENT_URL = '/api/pinned_event' as const;
 export const SOURCERER_API_URL = '/internal/security_solution/sourcerer' as const;
 
+/**
+ * This limit exists to maintain some kind of a safety net for how many events we are fetching in total,
+ * even though in theory we are only allowing up to 100 notes per document.
+ */
+export const NOTES_PER_PAGE_HARD_LIMIT = 10000;
```

Comment:
Do you think it makes sense to add API integration tests for the same. That will also demonstrate the performance of the API as well.

Reply:
it makes sense but the test would take a while to execute, wouldnt it?